### ISSUE TITLE
Colon rule doesn’t catch misplaced colons on closure parameters

### ISSUE THREAD

**Issue by ZevEisenberg** [realm/SwiftLint#2766]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The colon rule doesn’t catch misplaced colons on closure parameters.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files at paths 
Linting 'completion.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version: 0.32.0
* Installation method used: homebrew
* Paste your configuration file:

Not using a config file for this reduced test case.

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No.
* Which Xcode version are you using (check `xcode-select -p`)? Xcode 10.1 (10B61) on macOS 10.14.5 (18F132)
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`

```bash
cat completion.swift| swiftlint lint --no-cache --use-stdin --enable-all-rules
<nopath>:1:1: warning: Explicit ACL Violation: All declarations should specify Access Control Level keywords explicitly. (explicit_acl)
<nopath>:7:1: warning: Explicit ACL Violation: All declarations should specify Access Control Level keywords explicitly. (explicit_acl)
<nopath>:1:1: warning: Explicit Top Level ACL Violation: Top-level declarations should specify Access Control Level keywords explicitly. (explicit_top_level_acl)
<nopath>:7:1: warning: Explicit Top Level ACL Violation: Top-level declarations should specify Access Control Level keywords explicitly. (explicit_top_level_acl)
Done linting! Found 4 violations, 0 serious in 1 file.
```

```swift
// This does not trigger:
func foo(
    bar: String,
    baz: String,
    completion :@escaping (_ error: Error?) -> Void
    ) {}

func foo(
    bar: String,
    baz: String,
    completion :(_ error: Error?) -> Void
    ) {}
```

**Comment by phoney** [comment]
I ran into this one as well when running swiftlint autocorrect. I assume it's the same issue. Code like the following fixes the first parameter but not the second parameter.

``func foo(thisIsFixed:String, thisIsNotFixed:@escaping (_ error: Error?) -> Void) {
}``

**Comment by MadGeorge** [comment]
I think i have related case with simple property of closure type:
```swift
var eventHandler: (() -> Void)?
```
When formatted wrong
```swift
var eventHandler:(() -> Void)?
```
It will not be corrected with `swiftling autocorrect`

**Comment by ZevEisenberg** [comment]
@MadGeorge that enum one looks like an unrelated bug. Would you mind please filing a new issue for that one?

**Comment by ShadyGhalab** [comment]
> I think i have related case with simple property of closure type:
> 
> ```swift
> var eventHandler: (() -> Void)?
> ```
> 
> When formatted wrong
> 
> ```swift
> var eventHandler:(() -> Void)?
> ```
> 
> It will not be corrected with `swiftling autocorrect`

`colon` rule does not catch this for me. Have you figured it out?

**Comment by MadGeorge** [comment]
> `colon` rule does not catch this for me. Have you figured it out?

nope, i did fix it manually
